### DISCUSSION TITLE
How to retrive a specific tag?

### DISCUSSION THREAD

**Discussion by littlebear0871** [renanbr/bibtex-parser#100] [Q&A] [answered]
HI there!  I am trying to retrieve "doi" of each article, by using `foreach()` method. A error is reported:`Warning: Illegal string offset 'doi'.` following is the code:
```
foreach ($entries as $entrie) {
    foreach ($entrie as $bibitem) {
        echo $bibitem['doi'];
    }
```

Is any shortcut or method available to get certain tag?

Thakn you very much!

**Comment by renanbr** [comment, marked as answer]
The warning is about the missing tag in at least one of the entries.

To avoid this warning, you can either check if the tag is available before reading it ...

```php
$doi = $bibteim['doi'] ?? null; // PHP 7+
```

... or add the [FillMissingProcessor](https://github.com/renanbr/bibtex-parser#fill-missing-tag) to the Listener in order to make sure this key is always available.

**Reply by renanbr** [reply]
reviewing the snippet you shared, I would say it should look like this:

```php
foreach ($entries as $bibteim) {
    $doi = $bibteim['doi'] ?? null;
}
```

you're actually iterating over the tags in the inner foreach
